commit/merkleworker/types.go

```diff
@@ -0,0 +1,183 @@
+package commit
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/smartcontractkit/chainlink-ccip/plugintypes"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+)
+
+// TODO: Do we need this?
+type Query struct {
+	RmnOnRampMaxSeqNums []plugintypes.SeqNumChain
+	MerkleRoots         []cciptypes.MerkleRootChain
+}
+
+func (q Query) Encode() ([]byte, error) {
+	return json.Marshal(q)
+}
+
+type MerkleObservation struct {
+	MerkleRoots        []cciptypes.MerkleRootChain     `json:"merkleRoots"`
+	OnRampMaxSeqNums   []plugintypes.SeqNumChain       `json:"onRampMaxSeqNums"`
+	OffRampNextSeqNums []plugintypes.SeqNumChain       `json:"offRampNextSeqNums"`
+	FChain             map[cciptypes.ChainSelector]int `json:"fChain"`
+}
+
+func (obs MerkleObservation) Encode() ([]byte, error) {
```

Comment:
@makramkd @dimkouv what do you think about getting rid of every `Encode()` function in the plugins? Go allows us easily use reflection to do this in one place rather than sprinkling it throughout the types.

Reply:
Sounds good, we also want to replace this with proto soon